cli/src/registry/edit-with-tambo-button/edit-with-tambo-button.tsx

```diff
@@ -0,0 +1,407 @@
+  description?: string;
+  /** Optional className for the button */
+  className?: string;
+  /** Optional callback to open the thread panel/chat interface */
+  onOpenThread?: () => void;
+  /**
+   * Optional TipTap editor ref for inserting text when using "Send in Thread"
+   *
+   * NOTE: This implementation uses simple text insertion (setContent) to remain
+   * portable across different editor setups. It does NOT use TipTap Mention nodes
+   * or context attachments. If you need those features, implement them in your
+   * own wrapper or see apps/web/components/ui/tambo/edit-with-tambo-button.tsx for reference.
+   */
+  editorRef?: React.MutableRefObject<Editor | null>;
+  /** Optional suggestions to display when using "Send in Thread" */
+  suggestions?: Suggestion[];
+}
+
+/**
+ * Inline "Edit with Tambo" button and floating popover for interactable components.
+ * When clicked, opens a floating popover with a prompt input. Sends messages to Tambo
+ * with the interactable component in context and displays only the latest reply.
+ *
+ * Must be used within a component wrapped with `withInteractable`.
+ *
+ * @example
+ * ```tsx
+ * const MyInteractableForm = withInteractable(MyForm, {
+ *   componentName: "MyForm",
+ *   description: "A form component",
+ * });
+ *
+ * function MyForm() {
+ *   return (
+ *     <div>
+ *       <EditWithTamboButton />
+ *     </div>
+ *   );
+ * }
+ * ```
+ */
+
+export function EditWithTamboButton({
+  icon,
+  tooltip = "Edit with tambo",
+  description,
+  className,
+  onOpenThread,
+  editorRef,
+}: EditWithTamboButtonProps) {
+  const component = useTamboCurrentComponent();
+  const { sendThreadMessage, isIdle } = useTambo();
+
+  const [prompt, setPrompt] = useState("");
+  const [isPending, setIsPending] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const [sendMode, setSendMode] = useState<"send" | "thread">("send");
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const wasGeneratingRef = useRef(false);
+
+  // If no component, the current component is not an interactable - don't render.
+  if (!component) {
+    return null;
+  }
+
+  const isGenerating = !isIdle;
+
+  // Close popover when generation completes
+  useEffect(() => {
+    if (isGenerating) {
+      wasGeneratingRef.current = true;
+    } else if (wasGeneratingRef.current && !isPending) {
+      // Generation just completed - close the popover
+      wasGeneratingRef.current = false;
+      setIsOpen(false);
+      setPrompt("");
+    }
+  }, [isGenerating, isPending]);
+
+  const handleSend = useCallback(async () => {
+    if (!prompt.trim() || isPending) {
+      return;
+    }
+
+    setIsPending(true);
+    setError(null);
+
+    try {
+      // Send the message with the interactable component in context
+      await sendThreadMessage(prompt.trim(), {
```

Comment:
it feels a little strange that we're putting try/catch around this when we could be getting the pending/error/etc state from `useTambo()` - can we remove the pending/error management in this component for that?